docs/samples/dependency-injection/generic-host/Program.cs

```diff
@@ -0,0 +1,77 @@
+﻿using McMaster.Extensions.CommandLineUtils;
+using McMaster.Extensions.Hosting.CommandLine;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CustomServices
+{
+#region Program
+    [Command(Name = "di", Description = "Dependency Injection sample project")]
+    [HelpOption]
+    class Program
+    {
+        public static async Task<int> Main(string[] args)
+        {
+            return await new HostBuilder()
+                .ConfigureLogging((context, builder) =>
+                {
+                    builder.AddConsole();
+                })
+                .ConfigureServices((context, services) => {
+                    services.AddSingleton<IMyService, MyServiceImplementation>()
+                        .AddSingleton<IConsole>(PhysicalConsole.Singleton);
+                })
+                .RunCliAsync<Program>(args);
+        }
+
+        private readonly ILogger<Program> _logger;
+        private readonly IMyService _myService;
+
+        public Program(ILogger<Program> logger, IMyService myService)
+        {
+            _myService = myService;
+            _logger = logger;
+            
+            _logger.LogInformation("Constructed!");
+        }
+
+        private void OnExecute()
+        {
+            _myService.Invoke();
```

Comment:
Could you show an example of using `[Option]` or `[Argument]` as well?

Reply:
[Resolved](https://github.com/natemcmaster/CommandLineUtils/pull/167/commits/1bf26f3b37fcc6302667e3dea14dc936ba4d73f8)